pkg/webhook/v1alpha3/experiment/validator/validator.go

```diff
@@ -70,33 +63,13 @@ func (g *DefaultValidator) ValidateExperiment(instance *experimentsv1alpha3.Expe
 		return fmt.Errorf("Only one of spec.parameters and spec.nasConfig can be specified.")
 	}
 
-	if err := g.validateAlgorithmSettings(instance); err != nil {
```

Comment:
so do you think if we have some any method to validate AlgorithmSettings? how about implementing it in validateAlgorithm?

Reply:
Since we have suggestion deployment per experiment, we cannot do validate in web hook anymore. We can do a validate in controller itself and update status based on it. what do you think?

Reply:
sgtm

Reply:
Yes, we can do it within another PR, I think.

Reply:
Sounds good

Reply:
Yeah, today I will work on it and https://github.com/kubeflow/katib/pull/753#discussion_r324583651